include/picongpu/fields/incidentField/profiles/Wavepacket.hpp

```diff
@@ -46,45 +44,41 @@ namespace picongpu
                      * @tparam T_Params user (SI) parameters
                      */
                     template<typename T_Params>
-                    struct WavepacketUnitless : public T_Params
+                    struct WavepacketUnitless : public BaseTransversalGaussianParamUnitless<T_Params>
                     {
+                        //! User SI parameters
                         using Params = T_Params;
 
-                        static constexpr float_64 WAVE_LENGTH = Params::WAVE_LENGTH_SI / UNIT_LENGTH; // unit: meter
-                        static constexpr float_64 PULSE_LENGTH
-                            = Params::PULSE_LENGTH_SI / UNIT_TIME; // unit: seconds (1 sigma)
+                        //! Base unitless parameters
+                        using Base = BaseTransversalGaussianParamUnitless<T_Params>;
+
+                        // unit: UNIT_TIME
                         static constexpr float_X LASER_NOFOCUS_CONSTANT
-                            = float_X(Params::LASER_NOFOCUS_CONSTANT_SI / UNIT_TIME); // unit: seconds
-                        static constexpr float_64 AMPLITUDE = Params::AMPLITUDE_SI / UNIT_EFIELD; // unit: Volt /meter
-                        static constexpr float_X W0_AXIS_1
-                            = float_X(Params::W0_AXIS_1_SI / UNIT_LENGTH); // unit: meter
-                        static constexpr float_X W0_AXIS_2
-                            = float_X(Params::W0_AXIS_2_SI / UNIT_LENGTH); // unit: meter
-                        static constexpr float_64 INIT_TIME = Params::PULSE_INIT * PULSE_LENGTH
-                            + LASER_NOFOCUS_CONSTANT; // unit: seconds (full initialization length)
-                        static constexpr float_64 endUpramp = -0.5_X * LASER_NOFOCUS_CONSTANT; // unit: seconds
-                        static constexpr float_64 startDownramp = 0.5_X * LASER_NOFOCUS_CONSTANT; // unit: seconds
-                        static constexpr float_64 f = SPEED_OF_LIGHT / WAVE_LENGTH;
-                        static constexpr float_64 w = 2.0 * PI * f;
+                            = static_cast<float_X>(Params::LASER_NOFOCUS_CONSTANT_SI / UNIT_TIME);
+
+                        // unit: UNIT_TIME
+                        static constexpr float_X INIT_TIME

```

Comment:
Is the precision change to 32 bit intended?

Reply:
Same as above.